src/Doctrine/ManagerRegistry.php

```diff
@@ -29,7 +29,12 @@ class ManagerRegistry implements ManagerRegistryInterface
     /**
      * @var EntityManager[]
      */
-    protected $managers;
+    protected $originalManagers;
+
+    /**
+     * @var EntityManager[]
+     */
+    protected $resetManagers = [];
```

Comment:
Hey @dominikzogg 

Yes it works. I just added tests for this as proof.

Thanks,
Alex